plugins/api/include/irods/plugins/api/grid_configuration_types.h

```diff
@@ -0,0 +1,29 @@
+#ifndef IRODS_GRID_CONFIGURATION_TYPES_H
+#define IRODS_GRID_CONFIGURATION_TYPES_H
+
+#define IRODS_GRID_CONFIGURATION_COLUMN_SIZE 2700
+
+typedef struct GridConfigurationInput
+{
+    char name_space[IRODS_GRID_CONFIGURATION_COLUMN_SIZE];
+    char option_name[IRODS_GRID_CONFIGURATION_COLUMN_SIZE];
+    char option_value[IRODS_GRID_CONFIGURATION_COLUMN_SIZE];
+} gridConfigurationInp_t;
+
+#define GridConfigurationInp_PI "str name_space[2700]; "    \
+                                "str option_name[2700]; "   \
+                                "str option_value[2700];"
+//#define GridConfigurationInp_PI "str name_space[IRODS_GRID_CONFIGURATION_COLUMN_SIZE]; "    \
+//                                "str option_name[IRODS_GRID_CONFIGURATION_COLUMN_SIZE]; "   \
+//                                "str option_value[IRODS_GRID_CONFIGURATION_COLUMN_SIZE];"
```

Comment:
think about what to do with these

Reply:
Why change this from the constants?

Reply:
Packstruct doesn't know what that name is.

It requires updating the packing instruction mappings. 
https://github.com/irods/irods/blob/524314a6a8f0fbc067c16377b9efe034e341003d/lib/core/include/irods/rodsPackTable.h#L33-L58.
I didn't feel the name was appropriate for the packing instruction mappings.

I'd love to use that if we can come up with a more general name.

On the other hand, one could argue that maybe all fixed database column sizes should be mapped into packstruct. What would the convention be for those? `DATABASE_COLUMN_SIZE_2700`?

Reply:
Oooh, okay.

I like the idea of building the column sizes as constants into the packStruct. That seems generally useful. The name seems fine too. Perhaps a mention that it's a string since some of the columns are int, bigint, etc.? I think all of that could be a separate PR, by the way.

@trel, thoughts?

Reply:
I think we can hard code these values for now - and create a new issue for any abstractions.   want to get this in for 4.3.0 for now.

Reply:
This has been captured in an issue: #6297